scripts/snapshot/pre-snapshot-check.py

```diff
@@ -0,0 +1,61 @@
+import json
+import os
+
+import requests
+
+from utils import is_snapshot_week, get_dependency_version, get_latest_tag, get_snapshot_branch
+
+github_token = os.getenv("GITHUB_TOKEN")
+headers = {"Authorization": "Bearer " + github_token}
+
+
+def build_message():
+    message = '@navigation-ios '
+
+    releases_url = "https://api.github.com/repos/mapbox/mapbox-navigation-ios/releases"
+    releases = requests.get(releases_url).json()
+    if is_snapshot_week(releases):
+        message += 'Navigation SDK snapshot must be released today (rc or GA release was not released this week).\n'
+    else:
+        message += 'Navigation SDK snapshot must not be released today (rc or GA release was released this week).\n'
+        return message
+
+    maps_releases = requests.get(
+        'https://api.github.com/repos/mapbox/mapbox-maps-ios/releases',
+        headers=headers
+    ).json()
+    maps_version = get_dependency_version(maps_releases)
+    if maps_version:
+        message += ':white_check_mark: Maps ' + maps_version + ' is ready.\n'
+    else:
+        message += ':siren: Expected Maps release was not released.\n'
+
+    nav_native_releases = requests.get(
+        'https://api.github.com/repos/mapbox/mapbox-navigation-native/releases',
```

Comment:
Let's check [mapbox-navigation-native-ios](https://api.github.com/repos/mapbox/mapbox-navigation-native-ios) repo instead because some releases might be internal or not for the iOS platform.
NN pipeline can also take some time, so the native-ios repo can update with a delay.